src/Rules/Classes/EnumSanityRule.php

```diff
@@ -189,6 +181,20 @@ public function processNode(Node $node, Scope $scope): array
 			$exprType = $scope->getType($stmt->expr);
 			$scalarType = $enumNode->scalarType->toLowerString() === 'int' ? new IntegerType() : new StringType();
 			if ($scalarType->isSuperTypeOf($exprType)->yes()) {
+				$constantValues = $exprType->getConstantScalarValues();
```

Comment:
I guess you could support IntegerRanges with getFinitTypes()

Reply:
Do you have a real case scenario ?

EnumCase are support to be constant values. `case Foo = 1`
I'm just adding the support of thing like `case Foo = SomeClass::Constant` by not just looking at scalar int/string but directly the inferred type.

I don't see value of integer range here